packages/@storefront/core/src/services/tracker.ts

```diff
@@ -62,9 +63,20 @@ class TrackerService extends BaseService<TrackerService.Options> {
     }
   }
 
+  attachGbiEventMetadata = <S>(override: S): S => {
```

Comment:
@JoeyDeol 

Lets updating `S` to extend `BaseEvent`. ie.

```
<S extends GbTracker.BaseEvent>
```

Reply:
Potentially we should also investigate using a Union type in order to get over this problem.